File: samples/multi-project-plugin/server/src/main/java/example/server/ServerExampleBuildFeature.java

package example.server;

import jetbrains.buildServer.serverSide.BuildFeature;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

import static example.common.ExampleBuildFeature.BUILD_FEATURE_NAME;

public class ServerExampleBuildFeature extends BuildFeature {

    @NotNull
    @Override
    public String getType() {
        return BUILD_FEATURE_NAME;
    }

    @NotNull
    @Override
    public String getDisplayName() {
        return "Example Build Feature";
    }

    @Override
    public String getEditParametersUrl() {
        return null;
    }

    @Override
    public boolean isMultipleFeaturesPerBuildTypeAllowed() {
        return false;
    }

    @NotNull
    @Override
    public String describeParameters(@NotNull Map<String, String> params) {
        return "Example Build Feature plugin";
    }
}

File: samples/kotlin-plugin/server/src/main/kotlin/example/server/ServerExampleBuildFeature.kt

package example.server

import example.common.ExampleBuildFeature.BUILD_FEATURE_NAME
import jetbrains.buildServer.serverSide.BuildFeature

class ServerExampleBuildFeature : BuildFeature() {

    override fun getType(): String {
        return BUILD_FEATURE_NAME
    }

    override fun getDisplayName(): String {
        return "Example Build Feature"
    }

    override fun getEditParametersUrl(): String? {
        return null
    }

    override fun isMultipleFeaturesPerBuildTypeAllowed(): Boolean {
        return false
    }

    override fun describeParameters(params: Map<String, String>): String {
        return "Example Build Feature plugin"
    }
}
